rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -466,4 +480,48 @@ bool Player::publish_message(rosbag2_storage::SerializedBagMessageSharedPtr mess
   return message_published;
 }
 
+void Player::add_keyboard_callbacks()
+{
+  RCLCPP_INFO_STREAM(this->get_logger(), "Adding keyboard callbacks.");
+  // translated options into keybindings
+  const KeyboardHandler::KeyCode pause_resume_toggle_key = enum_str_to_key_code(
+    play_options_.pause_resume_toggle_key);
+  const KeyboardHandler::KeyCode play_next_key = enum_str_to_key_code(play_options_.play_next_key);
+  // check keybindings
+  if (pause_resume_toggle_key == KeyboardHandler::KeyCode::UNKNOWN) {
+    RCLCPP_ERROR_STREAM(
+      this->get_logger(),
+      "Invalid key binding " <<
+        play_options_.pause_resume_toggle_key <<
+        " for pause/resume toggle!");
+    throw std::invalid_argument("Invalid key binding.");
+  }
+  if (play_next_key == KeyboardHandler::KeyCode::UNKNOWN) {
+    RCLCPP_ERROR_STREAM(
+      this->get_logger(),
+      "Invalid key binding " <<
+        play_options_.play_next_key <<
+        " for play-next!");
+    throw std::invalid_argument("Invalid key binding.");
+  }
+  // create callback
+  KeyboardHandler::callback_t callback = [this, pause_resume_toggle_key, play_next_key](
+    KeyboardHandler::KeyCode key_code,
+    KeyboardHandler::KeyModifiers /*key_modifiers*/)
+    {
+      if (key_code == pause_resume_toggle_key) {
```

Comment:
stylistically - i would lean towards separate callbacks

```
keyboard_handler_.add_key_press_callback(KeyCode::SPACE, std::bind(Player::toggle_paused, this));
keyboard_handler_.add_key_press_callback(KeyCode::n, std::bind(Player::play_next, this));
```

And if not doing that, would lean towards a `switch` rather than `elif`-chain

Reply:
k done